### PR TITLE
Add alpine-arm64 to RID graph and use PackageReference for System.Data.SqlClient in tests again

### PR DESCRIPTION
**Pull request by akoeplinger**
Essentially reverts #31850 and fixes the underlying issue, which was that alpine-arm64 was missing from the RID graph.

This allows us to switch to PackageReference for SqlClient as the old approach doesn't work for mobile targets where we do self-contained publishing since we don't have the SqlClient dll in the in-tree runtime pack (and we don't want to put it there).
While working on this I also removed some unnecessary duplication between Mono and CoreCLR in runtime.depproj.

Also bumps Microsoft.DotNet.ProjectModel version in Microsoft.Extensions.DependencyModel.Tests.csproj

The older version references a non-existing assembly "System.Runtime.InteropServices.Pinvoke".

### CODE REVIEW

File: src/libraries/restore/runtime/runtime.depproj [hunk 1]
```diff
@@ -113,8 +93,7 @@
           AfterTargets="AfterResolveReferences;FilterNugetPackages"
           Condition="'$(RuntimeFlavor)' == 'Mono'">
     <ItemGroup>
-      <StoreTestHostFiles Include="@(ReferenceCopyLocalPaths)" />
-      <ReferenceCopyLocalPaths Remove="@(ReferenceCopyLocalPaths)" />
+      <!-- nothing to do here -->
```
Comment (ViktorHofer) [inline src/libraries/restore/runtime/runtime.depproj:96]: instead you can just close the `Target` element:

```
<Target .... />
```

Comment (akoeplinger) [reply src/libraries/restore/runtime/runtime.depproj]: Done.

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
@ericstj @wfurt can you please take a look at the rid change. Shouldn't prevent us from merging meanwhile though.

**Review by ViktorHofer — APPROVED** [review]

**Comment by akoeplinger** [comment]
I talked to @ericstj about this yesterday and he helped me track down the issue :)

The test failures are due to https://github.com/dotnet/runtime/issues/37946 and an issue with the Ubuntu.1804.Amd64.Open queue that I reported on Teams.

The important parts are the Alpine arm64 jobs which are green, merging.

File: src/libraries/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
     <RuntimeGroup Include="alpine">
       <Parent>linux-musl</Parent>
-      <Architectures>x64</Architectures>
+      <Architectures>x64;arm64</Architectures>
```
Comment (ericstj) [inline src/libraries/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props:18]: Do all these versions support arm64?  If not we could create a separate group for arm64.

Comment (akoeplinger) [reply src/libraries/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props:18]: Yes, all the version back to 3.6 support arm64.

**Pull request merged**